### PR TITLE
refs #756: fix PaymentMethod::getCurrentPaymentAmount

### PR DESCRIPTION
**Pull request by rvitaliy**
**Fixed issue**:  refs #756

### CODE REVIEW

**Comment by rvitaliy** [comment]
Travis CI fails, but it's not related to changes.
composer can't download Magento because authentication fails

**Comment by cyattilakiss** [comment]
Hi @rvitaliy ,

Thank you for opening this PR we are going to review this asap :)

Best,
Attila
Adyen

File: Helper/PaymentMethods.php [hunk 1]
```diff
@@ -252,15 +252,32 @@ protected function fetchAlternativeMethods($country)
         return $paymentMethods;
     }
 
-    /**
-     * @return bool|int
-     */
-    protected function getCurrentPaymentAmount()
+    protected function getCurrentPaymentAmount(): float
```
Comment (acampos1916) [inline Helper/PaymentMethods.php:255]: Supporting PHP<7
```suggestion
    protected function getCurrentPaymentAmount()
```

File: Helper/PaymentMethods.php [hunk 2]
```diff
@@ -252,15 +252,32 @@ protected function fetchAlternativeMethods($country)
...
     {
-        if (($grandTotal = $this->getQuote()->getGrandTotal()) > 0) {
+        $grandTotal = $this->getQuote()->getGrandTotal();
+
+        if (!\is_numeric($grandTotal)) {
```
Comment (acampos1916) [inline Helper/PaymentMethods.php:259]: I think there's no need to specify the global namespace for these functions?
```suggestion
        if (!is_numeric($grandTotal)) {
```

File: Helper/PaymentMethods.php [hunk 3]
```diff
@@ -252,15 +252,32 @@ protected function fetchAlternativeMethods($country)
...
+            throw new \Exception(
+                \sprintf(
```
Comment (acampos1916) [inline Helper/PaymentMethods.php:261]: I think there's no need to specify the global namespace for these functions?
```suggestion
                sprintf(
```

File: Helper/PaymentMethods.php [hunk 4]
```diff
@@ -252,15 +252,32 @@ protected function fetchAlternativeMethods($country)
...
+
+        $grandTotal = (float) $grandTotal;
+
+        if ($grandTotal > 0.0) {
             return $grandTotal;
         }
-        return 10;
+
+        throw new \Exception(
+            \sprintf(
```
Comment (acampos1916) [inline Helper/PaymentMethods.php:275]: I think there's no need to specify the global namespace for these functions?
```suggestion
            sprintf(
```

**Comment by rvitaliy** [comment]
Hi @cyattilakiss!
I seen your last 2 commits.
The result is same, but if we work with **float** type we must compare with **float** and not with int.
PHP allow us to do it, but it's theoretically wrong :) 

It's just a note to keep in mind.
If you prefer we can keep these changes.

**Comment by cyattilakiss** [comment]
> Hi @cyattilakiss!
> I seen your last 2 commits.
> The result is same, but if we work with **float** type we must compare with **float** and not with int.
> PHP allow us to do it, but it's theoretically wrong :)
> 
> It's just a note to keep in mind.
> If you prefer we can keep these changes.

Hey @rvitaliy 
Thank you for the response. This time I prefer to keep it and not cast the zero to float for only the comparison. (PHP will do it anyway in the background)
Thanks and now we are going to merge it finally :)
Best,
Attila

**Pull request merged**